tfjs-backend-webgpu/src/matmul_packed_webgpu.ts

```diff
@@ -33,21 +33,27 @@ export function matMulReadFnSource(
       let batchASize = uniforms.aShape[1] * uniforms.aShape[2];
       ${
       transposeA ?
-          `value = A[(batch * batchASize + col * uniforms.aShape[2] + row) / ${
-              component}];` :
-          `value = A[(batch * batchASize + row * uniforms.aShape[2] + col) / ${
-              component}];`}
+          `value = ${
+              typeSnippet(
+                  component)}(A[(batch * batchASize + col * uniforms.aShape[2] + row) / ${
```

Comment:
Please compare the performance impact on below two methods and choose the best one:
1.  use `${typeSnippet(component)}(A[(batch * batchASize + col **uniforms.aShape[2] + row)`
2. use `value = getA(batch, col, row)`.

Reply:
Can we re-visit the precision issue caused by casting again? Say

```typescript
const a = tf.tensor2d([19260817], [1, 1], 'int32');
const b = tf.tensor2d([1], [1, 1], 'int32');
const c = tf.matMul(a, b);  // 19260816, using this patch
                            // 19260817, with cpu backend
```

While precision loss is expected for float32 operands, I feel the behavior will surprise people in int32-only cases. In an ideal world where all calculations are performed using the "common type" (i32 here) in shader, such an issue could have been avoided.

Reply:
@qjia7 After the investigation, helper function `getA` does not win or lose performance (96%~102%) on CFL and TGL.
Updated shader maker to use helper function, PTAL.

  | mobileNetV3 (42) |   | movenet-singlepose(49) |   | DeepLabV3-pascal(37) |  
-- | -- | -- | -- | -- | -- | --
  | origin | getX | origin | getX | origin | getX
CFL | 6.34 | 6.88 | 13.29 | 13.32 | 53.19 | 53.32
  | 6.31 | 5.77 | 15.17 | 14.14 | 50.98 | 53.27
  | 5.86 | 6.31 | 15.11 | 14.21 | 52.95 | 50.93
TGL | 3.73 | 3.71 | 7.67 | 8.3 | 13.93 | 14.09
  | 3.6 | 3.72 | 8.29 | 8.22 | 13.95 | 13.68
  | 3.73 | 3.72 | 8.36 | 7.8 | 13.69 | 13.72
  |   |   |   |   |   |  
CFL_AVE | 6.17 | 6.32 | 14.52 | 13.89 | 52.37 | 52.51
TGL_AVE | 3.69 | 3.72 | 8.11 | 8.11 | 13.86 | 13.83
CFL_DIFF | 102% | 96% | 100%
TGL_DIFF | 101% | 100% | 100%